beacon-chain/core/altair/sync_committee.go

```diff
@@ -221,3 +242,85 @@ func SubnetsForSyncCommittee(state iface.BeaconStateAltair, i types.ValidatorInd
 func SyncCommitteePeriod(epoch types.Epoch) uint64 {
 	return uint64(epoch / params.BeaconConfig().EpochsPerSyncCommitteePeriod)
 }
+
+// SyncSubCommitteePubkeys returns the pubkeys participating in a sync subcommittee.
+//
+// def get_sync_subcommittee_pubkeys(state: BeaconState, subcommittee_index: uint64) -> Sequence[BLSPubkey]:
+//    # Committees assigned to `slot` sign for `slot - 1`
+//    # This creates the exceptional logic below when transitioning between sync committee periods
+//    next_slot_epoch = compute_epoch_at_slot(Slot(state.slot + 1))
+//    if compute_sync_committee_period(get_current_epoch(state)) == compute_sync_committee_period(next_slot_epoch):
+//        sync_committee = state.current_sync_committee
+//    else:
+//        sync_committee = state.next_sync_committee
+//
+//    # Return pubkeys for the subcommittee index
+//    sync_subcommittee_size = SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT
+//    i = subcommittee_index * sync_subcommittee_size
+//    return sync_committee.pubkeys[i:i + sync_subcommittee_size]
+func SyncSubCommitteePubkeys(st iface.BeaconStateAltair, subComIdx types.CommitteeIndex) ([][]byte, error) {
```

Comment:
We really dont need this. It can get expensive for every incoming frequent object `sync_committee_contribution_and_proof`. You can use the existing `sync_committee` cache to accomplish the same